hal/src/nRF52840/pinmap_hal.c

```diff
@@ -15,16 +15,14 @@
  * License along with this library; if not, see <http://www.gnu.org/licenses/>.
  */
 
-/* Includes ------------------------------------------------------------------*/
 #include "pinmap_hal.h"
 #include "pinmap_impl.h"
 #include "pwm_hal.h"
 #include "module_info.h"
 
-NRF5x_Pin_Info __PIN_MAP[TOTAL_PINS] =
-{
-#if PLATFORM_ID == PLATFORM_ARGON || PLATFORM_ID == PLATFORM_BORON || PLATFORM_ID == PLATFORM_XENON
+#if PLATFORM_ID == PLATFORM_ARGON
```

Comment:
Should we move these to platform-specific source files as well?

Reply:
I'll split `pinmap_hal.c` and `pinmap_impl.h` into each platform folders, should we rename the pin definition file with the name `pinmap_defines.h` which will be included in `pinmap_impl.h`?